### DISCUSSION TITLE
Moving consumed but unmaintained packages out of the way

### DISCUSSION THREAD

**Discussion by peterwilsoncc** [WordPress/gutenberg#52815] [Developer Experience]
In #52444, `@wordpress/nux` was restored to the Gutenberg repository as the package is still being consumed by WordPress-Develop for the WordPress release packages.

Unfortunately, without version bumps the syncing script @gziolo and @adamziel worked on for WordPress releases fails and downgrades packages to versions intended for previous versions of WordPress.

To avoid clogging up the Gutenberg `/packages` directory with unmaintained packages but consumed packages, I propose moving such packages to another location. Lerna can be configured to handle [multiple package locations](https://github.com/lerna/lerna/issues/126).

While adding the additional directory to the lerna config appears to be straight forward, various scripts for the build and testing process would need to be updated: `clean:packages` and `./bin/packages/*` for example.

This only need apply to consumed packages, once they are no longer consumed (or stubbed without any dependencies) they can be removed.

Is it worth the development effort to modify the various scripts needed to build the packages to move the unmaintained packages out of the way?

CC @youknowriad @gziolo @mcsf @noisysocks @priethor @aristath @adamziel @carolinan @tellthemachines @talldan @georgeh @artemiomorales @kevin940726 @ramonjd @andrewserong @ellatrix (I copied this list from https://github.com/WordPress/gutenberg/discussions/46693, let me know I others ought to be pinged)

**Comment by gziolo** [comment]
> To avoid clogging up the Gutenberg /packages directory with unmaintained packages but consumed packages, I propose moving such packages to another location. Lerna can be configured to handle https://github.com/lerna/lerna/issues/126.

Yes, that would work, but it might impact a few existing scripts that control things like build, types, or documentation. It would have to get explored.

**Reply by adamziel** [reply]
Yup, assessing the surface of this would be a micro project in itself.
